src/main/java/org/mskcc/smile/service/impl/ValidRequestCheckerImpl.java

```diff
@@ -72,24 +73,36 @@ public String getFilteredValidRequestJson(String requestJson) throws IOException
             } else {
                 sampleStatus = generateNonCmoSampleValidationMap(sampleMap);
             }
+            sampleMap.put("status", sampleStatus);
+            Object sampleObj = mapper.convertValue(sampleMap, Object.class);
+
+            // get validation report and check status
+            Map<String, String> validationReport =
+                    mapper.convertValue(sampleStatus.get("validationReport"), Map.class);
             if ((Boolean) sampleStatus.get("validationStatus")) {
                 validSampleCount++;
+                updatedSampleList.add(sampleObj);
+            } else {
+                // do not add samples from cmo request if they are missing cmo patient ids
+                if (validationReport.containsKey("cmoPatientId")) {
+                    LOG.warn("Adding CMO sample with missing CMO patient ID to request-level "
+                            + "validation report (failed samples): " + mapper.writeValueAsString(sampleMap));
+                    invalidRequestSamplesStatuses.add(sampleObj);
+                } else {
+                    updatedSampleList.add(sampleObj);
+                }
             }
-            sampleMap.put("status", sampleStatus);
-            updatedSampleList.add(mapper.convertValue(sampleMap, Object.class));
         }
-        // update request status 'validationStatus' based on results from sample validation
+        // update request json with request status and samples containing validation reports
         if ((Boolean) requestStatus.get("validationStatus")) {
-            List<String> requestValidationReport = (List<String>) requestStatus.get("validationReport");
+            Map<String, Object> requestValidationReport = new HashMap<>();
             if (validSampleCount == 0) {
-                requestValidationReport.add("all request samples failed validation");
                 requestStatus.replace("validationStatus", Boolean.FALSE);
-            } else if (validSampleCount < sampleList.length) {
-                requestValidationReport.add("some request samples failed validation");
             }
+            requestValidationReport.put("samples",
```

Comment:
Need to fix this so that an empty list of samples isn't added to the `validationReport` when request passes validation